rpcs3/rpcs3qt/gl_gs_frame.cpp

```diff
@@ -29,9 +30,12 @@ draw_context_t gl_gs_frame::make_context()
 
 	if (m_primary_context)
 	{
-		auto surface = new QOffscreenSurface();
+		QOffscreenSurface* surface = new QOffscreenSurface();
 		surface->setFormat(m_format);
-		surface->create();
+		Emu.BlockingCallFromMainThread([&]()
+		{
+			surface->create();
```

Comment:
Why? You can hand over opengl contexts between threads using glMakeCurrent

Reply:
In fact this is why the set_current method exists. The thread calling set_current "acquires" the gl render context. Such hacks aren't necessary.

Reply:
This [note](https://doc.qt.io/qt-5/qoffscreensurface.html#details) is from the Qt documentation.

> Note: Due to the fact that `QOffscreenSurface` is backed by a [`QWindow`](https://doc.qt.io/qt-5/qwindow.html) on some platforms, cross-platform applications must ensure that [`create`](https://doc.qt.io/qt-5/qoffscreensurface.html#create)() is only called on the main (GUI) thread. The `QOffscreenSurface` is then safe to be used with [`makeCurrent()`](https://doc.qt.io/qt-5/qopenglcontext.html#makeCurrent) on other threads, but the initialization and destruction must always happen on the main (GUI) thread.

Reply:
Just the kind of hacky BS one should expect from Qt xD

Reply:
Add a comment explaining that this anti-pattern is a workaround to a Qt issue. That way it doesn't get removed in future.

Reply:
I added a comment.